spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -98,17 +98,27 @@
   private Filter[] pushedFilters = NO_FILTERS;
 
   SparkScanBuilder(
-      SparkSession spark, Table table, Schema schema, CaseInsensitiveStringMap options) {
+      SparkSession spark,
+      Table table,
+      String branch,
+      Schema schema,
+      CaseInsensitiveStringMap options) {
     this.spark = spark;
     this.table = table;
     this.schema = schema;
     this.options = options;
-    this.readConf = new SparkReadConf(spark, table, options);
+    this.readConf = new SparkReadConf(spark, table, branch, options);
     this.caseSensitive = readConf.caseSensitive();
   }
 
-  SparkScanBuilder(SparkSession spark, Table table, CaseInsensitiveStringMap options) {
-    this(spark, table, table.schema(), options);
+  SparkScanBuilder(
+      SparkSession spark, Table table, String branch, CaseInsensitiveStringMap options) {
+    this(spark, table, branch, SnapshotUtil.schemaFor(table, branch), options);
+  }
+
+  SparkScanBuilder(
```

Comment:
Is it possible to avoid needing to cast `(String) null` in tests? Maybe one of these could be private?

Reply:
I added back constructor 

```
SparkScanBuilder(SparkSession spark, Table table, CaseInsensitiveStringMap options) {
    this(spark, table, table.schema(), options);
  }
```

to avoid this